instrumentation/rails/lib/opentelemetry/instrumentation/rails/instrumentation.rb

```diff
@@ -22,6 +22,8 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
           defined?(::Rails)
         end
 
+        option :enable_recognize_route, default: false, validate: :boolean
```

Comment:
I went back and forth on wether or not this was the right name for this setting, typically I wouldn't want to associate the name with the implementation (ie. `router.recognize`).  But in the case I think this makes sense as the reason we are putting this behind a config is due to performance concerns.